sqlglot/parser.py

```diff
@@ -2565,6 +2567,24 @@ def _parse_select_or_expression(self):
     def _parse_use(self):
         return self.expression(exp.Use, this=self._parse_id_var())
 
+    def _parse_transaction(self):
+        this = None
+        if (
+            self._match_text("DEFERRED")
+            or self._match_text("IMMEDIATE")
+            or self._match_text("EXCLUSIVE")
```

Comment:
I don't like this much. Maybe a better approach would be to change `_match_text` so that it can check against a set of strings (kwarg) for a _single_ match.

Reply:
i guess add a new one that does an in clause